tests/queries/0_stateless/02908_many_requests_to_system_replicas.sh

```diff
@@ -7,8 +7,8 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 
 set -e
 
-NUM_TABLES=300
-CONCURRENCY=200
+NUM_TABLES=50
```

Comment:
Would it make sense to add a function like `is_built_with_sanitizer`  in integration tests and reduce the number of tables/concurrency based on it?
Not just for this case but could be useful in general if we want to avoid disabling a test in sanitizer builds completely or to avoid reducing complexity for release builds